### ISSUE TITLE
Dune not found on docker ?

### ISSUE THREAD

**Issue by waghanza** [aantron/dream#363]
Hi @aantron,

With the following dockerfile
```dockerfile
FROM ocaml/opam:debian-12-ocaml-5.2

WORKDIR /usr/src/app

USER root
RUN apt-get update -y && apt-get install pkg-config libev-dev libssl-dev libgmp-dev -y
RUN apt-get -qy install curl
USER opam

  COPY 'server.opam' 'server.opam'
  COPY 'server.ml' 'server.ml'
  COPY 'dune' 'dune'
  COPY 'dune-project' 'dune-project'

  RUN opam install --deps-only --yes .



  ENTRYPOINT dune exec --root . ./server.exe

HEALTHCHECK CMD curl --fail http://0.0.0.0:3000 || exit 1
```

I can not run `dream` => https://github.com/the-benchmarker/web-frameworks/blob/463af92e3299278d4d2db055772cab3711837a58/ocaml/dream/server.ml#L1

I have an error 
```
$ dune exec --root . ./server.exe
bash: dune: command not found
```

Any idea ?

Regards,

**Comment by aantron** [comment]
Do you have the full log for this? If so, can you link me to it or post it? Was it working before?

**Comment by waghanza** [comment]
The full log of what ? opam ? How can I create a log of this package manager ?

Yes, it was working before, maybe an image update

**Comment by aantron** [comment]
The build of this container. It might also be good to add `--verbose` to `opam install --deps-only --yes .`

This is the repo for the team that generates the base Docker images. For our reference in the future of this convo if this starts to look like a base image issue: https://github.com/ocurrent/docker-base-images

**Comment by waghanza** [comment]
Full log is here https://gist.github.com/waghanza/41091125622c487c7aea783d0c96b8d7 @aantron

**Comment by aantron** [comment]
@mtelvers do you know what could cause this?

**Comment by mtelvers** [comment]
The `ENTRYPOINT` should read:
```
ENTRYPOINT opam exec -- dune exec --root . ./server.exe
```

TL;DR `opam` installs the `dune` binary in the switch `bin` folder: `/home/opam/.opam/5.2/bin` in your case.  In an interactive session, we would run `eval $(opam env)` to update the PATH variable.  For non interactive sessions, we can use `opam exec -- foo` to run command `foo` in the correct environment:

```
opam@f8e6556fa561:/usr/src/app$ whereis -b dune
dune:
opam@f8e6556fa561:/usr/src/app$ opam exec -- whereis -b dune
dune: /home/opam/.opam/5.2/bin/dune
```

**Comment by aantron** [comment]
Thank you @mtelvers!
